Source/Core/Core/AchievementManager.cpp

```diff
@@ -716,7 +716,10 @@ void AchievementManager::HandleLeaderboardTriggeredEvent(const rc_runtime_event_
   {
     if (m_game_data.leaderboards[ix].id == runtime_event->id)
     {
-      OSD::AddMessage(fmt::format("Scored {} on leaderboard: {}", runtime_event->value,
+      FormattedValue value{};
+      rc_runtime_format_lboard_value(value.data(), static_cast<int>(value.size()),
+                                     runtime_event->value, m_game_data.leaderboards[ix].format);
+      OSD::AddMessage(fmt::format("Scored {} on leaderboard: {}", value.data(),
```

Comment:
If we are concerned with null-termination, we could at least clamp if we don't find it in the buffer.  Something like:

```
if (std::find(value.begin(), value.end(), '\0') == value.end())
{
  OSD::AddMessage(fmt::format("Scored {} on leaderboard: {}", std::string_view{value.data(), value.size()}, ...
}
else
{
  OSD::AddMessage(fmt::format("Scored {} on leaderboard: {}", value.data(), ...
}
```